### ISSUE TITLE
[Bug]: SIP/2.0 400 Bad Request

### ISSUE THREAD

**Issue by satirebird** [BelledonneCommunications/linphone-desktop#789]
### Context

If linphone tries to register my SIP account (telekom) it receives a 400 Bad Request from the server. I captured it with wireshark. My other provider sipgate works well. 

Request:
```
REGISTER sip:tel.t-online.de SIP/2.0
Via: SIP/2.0/UDP 192.168.3.78:5060;branch=z9hG4bK.ROY3k-PgO;rport
From: <sip:000000000000@tel.t-online.de>;tag=ja~vy-zRe
To: sip:000000000000@tel.t-online.de
CSeq: 41 REGISTER
Call-ID: utZ42PoDVN
Max-Forwards: 70
Supported: replaces, outbound, gruu, record-aware
Accept: application/sdp
Accept: text/plain
Accept: application/vnd.gsma.rcs-ft-http+xml
Contact: <sip:000000000000@192.168.3.78;transport=udp>;message-expires=604800;+sip.instance="<urn:uuid:33dd8428-32bf-00c9-8a18-b0e619e40bb7>";+org.linphone.specs="ephemeral/1.1,groupchat/1.2,lime"
Expires: 600
User-Agent: Linphone-Desktop/5.1.2 (alfred) ubuntu/20.04 Qt/5.15.2 LinphoneSDK/5.2.97
```

Answer:
```
SIP/2.0 400 Bad Request
Via: SIP/2.0/UDP 192.168.3.78:5060;branch=z9hG4bK.ROY3k-PgO;rport
From: <sip:000000000000@tel.t-online.de>;tag=ja~vy-zRe
To: sip:000000000000@tel.t-online.de;tag=SipLib_02A51670BA30-7bc5-0-a1a1cf-650c27a9-3fd90
Call-ID: utZ42PoDVN
CSeq: 41 REGISTER
Content-Length: 0
```

Whats going wrong here?

### General information

- Device: PC
- OS: Ubuntu 20.04
- Version of the App: 5.1.2 (AppImage)

### Expected behaviour

Successful registration

### To Reproduce

1. Configure a SIP account

**Comment by codiflow** [comment]
I got the same problem with Linphone and the German "Telekom" as VoIP provider. Seems like this has to do with the name resolution of tel.t-online.de as stated here: https://support.securepoint.de/viewtopic.php?t=9843

They do not resolve A records for their domain since I think 2021 so you have to do SRV-requests to get the IP of the server.

I will report back as soon as I found a solution for this.

**Comment by codiflow** [comment]
The solution was to completely remove the accounts from Linphone and add them through the assistant instead of through the settings dialog.

![grafik](https://github.com/BelledonneCommunications/linphone-desktop/assets/43847817/fb99936f-75ae-4961-8e32-9aab4647e49b)

Linphone itself can resolve the SRV entries from DNS just fine. In case of problems just look at the log files within Linphone – they should help you.

**Comment by codiflow** [comment]
I wrote a step-by-step tutorial (only in German) for the process of using Telekom SIP accounts with Linphone today:
https://andersgood.de/kurz-notiert/voip-telefonie-der-telekom-mit-linphone-nutzen

**Comment by codiflow** [comment]
Does this already help you? It's the log of the complete registration process with all REGISTER messages:
https://andersgood.de/kurz-notiert/voip-telefonie-der-telekom-mit-linphone-nutzen#vollstaendiger-authentifizierungsprozess

**Comment by satirebird** [comment]
I had some time to create a small python script that sends the initial SIP register message to the telekom server. I figured out that the following parameter makes the difference:
```
+org.linphone.specs="ephemeral/1.1,groupchat/1.2,lime"
```
It is the last part of the `Contact` parameter. If I change it to 
```
+org.linphone.specs="lime"
```
the server accepts it. In your log only the "lime" spec is listed.

Currently I'm not sure which linphone settings affects this thing. Do you have an idea?

**Comment by codiflow** [comment]
Unfortunately no.

I also don't have the respective client with the Telekom server configured at hand so I unfortunately can't check your investigation results against my config.

**Comment by petervnv** [comment]
> I had some time to create a small python script that sends the initial SIP register message to the telekom server. I figured out that the following parameter makes the difference:
> 
> ```
> +org.linphone.specs="ephemeral/1.1,groupchat/1.2,lime"
> ```
> 
> It is the last part of the `Contact` parameter. If I change it to
> 
> ```
> +org.linphone.specs="lime"
> ```
> 
> the server accepts it. In your log only the "lime" spec is listed.
> 
> Currently I'm not sure which linphone settings affects this thing. Do you have an idea?

You can set the linphone_specs parameters in linphonerc

[sip]
linphone_specs=groupchat,lime

Don't know if it works with the desktop client. It doesn't seem to work with the iOS client as reported [here](https://github.com/BelledonneCommunications/linphone-iphone/issues/909)

Hope this helps

**Comment by satirebird** [comment]
> > I had some time to create a small python script that sends the initial SIP register message to the telekom server. I figured out that the following parameter makes the difference:
> > ```
> > +org.linphone.specs="ephemeral/1.1,groupchat/1.2,lime"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It is the last part of the `Contact` parameter. If I change it to
> > ```
> > +org.linphone.specs="lime"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the server accepts it. In your log only the "lime" spec is listed.
> > Currently I'm not sure which linphone settings affects this thing. Do you have an idea?
> 
> You can set the linphone_specs parameters in linphonerc
> 
> [sip] linphone_specs=groupchat,lime
> 
> Don't know if it works with the desktop client. It doesn't seem to work with the iOS client as reported [here](https://github.com/BelledonneCommunications/linphone-iphone/issues/909)
> 
> Hope this helps

You are right. It's listed with the groupchat and ephemeral enabled in the linphone_specs. But even if I change or remove it, linphone re-writes the original content. 
Does anyone how to disable it?

**Comment by codiflow** [comment]
As far as I remember Linphone has some kind of backup/integrity check mechanism regarding manual config changes.

You have to keep Linphone closed completely (not sent to tray) before you can edit the config.

Also look for a config "backup" file in the config directory and also update this.
